Sources/Private/CoreAnimation/Animations/EllipseAnimation.swift

```diff
@@ -14,15 +14,44 @@ extension CAShapeLayer {
   {
     try addAnimation(
       for: .path,
-      keyframes: ellipse.size.keyframes,
-      value: { sizeKeyframe in
+      keyframes: ellipse.combinedKeyframes(context: context).keyframes,
+      value: { keyframe in
         BezierPath.ellipse(
-          size: sizeKeyframe.sizeValue,
-          center: try ellipse.position.exactlyOneKeyframe(context: context, description: "ellipse position").value.pointValue,
+          size: keyframe.size.sizeValue,
+          center: keyframe.position.pointValue,
           direction: ellipse.direction)
           .cgPath()
           .duplicated(times: pathMultiplier)
       },
       context: context)
   }
 }
+
+extension Ellipse {
+  /// Data that represents how to render an ellipse at a specific point in time
+  struct Keyframe {
+    let size: Vector3D
+    let position: Vector3D
+  }
+
+  /// Creates a single array of animatable keyframes from the separate arrays of keyframes in this Ellipse
+  func combinedKeyframes(context: LayerAnimationContext) throws-> KeyframeGroup<Ellipse.Keyframe> {
+    let combinedKeyframes = Keyframes.combinedIfPossible(size, position) { untypedValues in
+      Keyframe(
+        size: untypedValues[0] as! Vector3D,
+        position: untypedValues[1] as! Vector3D)
+    }
```

Comment:
I'm a bit confused what's going on here, why are we force casting these to `Vector3D` if both params are of type `Vector3D`? Is this due to limitations around variadic parameters and generics? Wondering if we could instead just make multiple arities of this function with tuples, similar to how Combine works with operators, since this feels a bit brittle/unsafe as-is

Reply:
Looks like you answered this before I refreshed the page, feel free to merge as-is if you'd like 👍

Reply:
Having multiple arities is an interesting idea, I'll try it out and see how it goes. It would definitely be easier to understand the callsite without this casting.